protocol/packages/swap/src/astroport/mod.rs

```diff
@@ -33,37 +31,17 @@ mod testing;
 type RequestMsg = MsgExecuteContract;
 type ResponseMsg = MsgExecuteContractResponse;
 
-trait Router {
-    const ROUTER_ADDR: &'static str;
-}
-
-pub struct Main {}
-
-impl Router for Main {
-    /// Source: https://github.com/astroport-fi/astroport-changelog/blob/main/neutron/neutron-1/core_mainnet.json
-    const ROUTER_ADDR: &'static str =
-        "neutron1rwj6mfxzzrwskur73v326xwuff52vygqk73lr7azkehnfzz5f5wskwekf4";
-}
-
-pub struct Test {}
-
-impl Router for Test {
-    /// Source: https://github.com/astroport-fi/astroport-changelog/blob/main/neutron/pion-1/core_testnet.json
-    const ROUTER_ADDR: &'static str =
-        "neutron12jm24l9lr9cupufqjuxpdjnnweana4h66tsx5cl800mke26td26sq7m05p";
-}
-
 #[derive(Serialize, Deserialize)]
-pub struct RouterImpl<R>(PhantomData<R>);
+pub enum Impl
+where
+    Self: ExactAmountIn, {}
```

Comment:
[all dex-es] We agreed on bounding to `Self: Router`

Reply:
Yes, I remember that we did but the compiler/clippy complained that the trait was more private. Should I just make it public too?

Reply:
nope, let's try with a where clause at `impl ExactAmountIn for Impl ` would that work?